scripts/exportTypes.js

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+This little utility recognizes the TypeScript types defined in the input folder
+and automatically updates the exports in the specified output file.
+
+usage:
+
+node node scripts/exportTypes ./src ./src/index.ts types.d.ts index.ts
```

Comment:
Normally, this wouldn't 100% be the case.  It's not uncommon to export a method from a class for usage with tests or other classes in the project, but not want to make that externally available.  I think I understand the problem you're trying to solve for, but this generally makes me nervous.

Reply:
@JustinBeckwith Whoops, I meant to make this a draft.

I put this together just for myself then thought it might work in the library, but your feedback is exactly what I was hoping to get re: cases that aren't currently in this library, but might come about. Sounds like the nervousness is valid and we'll just have to be mindful of manually updating exports. Thank you!